lib/animation/animation.dart

```diff
@@ -0,0 +1,30 @@
+part of supercharged;
+
+/// Supercharged extension on [Animation] objects.
+extension Animation_ on Animation {
+  /// Returns an [Animation] after appling [CurveTween] based on begin and end.
+  /// Optionally can passe [Curve] as well by pass curve object at last parameter.
+  ///
+  /// Example:
+  /// ```dart
+  /// controller.interval(0.0, 1.0,curve:Curves.elasticOut)
+  /// ```
+  Animation interval(double begin, double end, {Curve curve}) {
+    // Value Guard
```

Comment:
I don't think we need that "value guard" comment. `assert`s should be known to developers.